### PR TITLE
docker: Use variable for release label value in dockerfile

### PR DESCRIPTION
**Pull request by hershpa**
Use variable ${KERNEL_FULL_VERSION} for release label
Signed-off-by: Hersh Pathak <hersh.pathak@intel.com>

### CODE REVIEW

**Comment by hershpa** [comment]
Hi @chaitanya1731 and @uMartinXu, can you please help review this PR?

**Comment by chaitanya1731** [comment]
@hershpa @uMartinXu  can we update the kernel version in the release [readme](https://github.com/intel/intel-data-center-gpu-driver-for-openshift/tree/main/release#table-1-intel-data-center-gpu-driver-container-images-for-openshift-release) as well in the same PR, just to keep track of it in the same place?

**Comment by uMartinXu** [comment]
> @hershpa @uMartinXu can we update the kernel version in the release [readme](https://github.com/intel/intel-data-center-gpu-driver-for-openshift/tree/main/release#table-1-intel-data-center-gpu-driver-container-images-for-openshift-release) as well in the same PR, just to keep track of it in the same place?

We should do that. But the release readme is for the released image. So I think only after we published the image. we can then update it.  Now the contends in the table is only an example.

**Comment by hershpa** [comment]
> > @hershpa @uMartinXu can we update the kernel version in the release [readme](https://github.com/intel/intel-data-center-gpu-driver-for-openshift/tree/main/release#table-1-intel-data-center-gpu-driver-container-images-for-openshift-release) as well in the same PR, just to keep track of it in the same place?
> 
> We should do that. But the release readme is for the released image. So I think only after we published the image. we can then update it. Now the contends in the table is only an example.

In that case, I think this PR is ready to merge.

**Comment by hershpa** [comment]
Can we merge this PR or is it on hold?

File: docker/intel-dgpu-driver.Dockerfile [hunk 1]
```diff
@@ -33,4 +33,4 @@ COPY --from=builder /etc/driver-toolkit-release.json /etc/
 COPY --from=builder /lib/modules/$KERNEL_FULL_VERSION/ /opt/lib/modules/$KERNEL_FULL_VERSION/
 COPY --from=builder /build/intel-gpu-firmware/firmware/ /firmware/i915/
 
-RUN depmod -b /opt
```
Comment (uMartinXu) [inline docker/intel-dgpu-driver.Dockerfile:36]: why we have this line? :-)

**Comment by uMartinXu** [comment]
> @hershpa @uMartinXu can we update the kernel version in the release [readme](https://github.com/intel/intel-data-center-gpu-driver-for-openshift/tree/main/release#table-1-intel-data-center-gpu-driver-container-images-for-openshift-release) as well in the same PR, just to keep track of it in the same place?

I think the table in the release readme should be updated after the image is published. Right?

File: docker/intel-dgpu-driver.Dockerfile (hunk 1 above)
Comment (hershpa) [reply docker/intel-dgpu-driver.Dockerfile:36]: That line runs depmod from the /opt directory in the final driver container image. It creates the modules.dep file that creates a dependency list for each driver. For example, i915 requires intel_vsec to be loaded first. I believe this is required. During the build, a depmod is run but it is in the default /lib directory. We had added this line based on the KMM examples: https://openshift-kmm.netlify.app/lab/#in-cluster-build-from-sources-single-step.

Comment (hershpa) [reply docker/intel-dgpu-driver.Dockerfile:36]: COPY --from=builder /etc/driver-toolkit-release.json /etc/: This line is optional, it simply adds the driver-toolkit version in a file for reference. We can remove it in a separate PR. It was something we inherited from early SRO/KMM examples.

**Comment by hershpa** [comment]
@chaitanya1731  and @uMartinXu, good to merge?

**Comment by hershpa** [comment]
Just following up on previous question.

File: docker/intel-dgpu-driver.Dockerfile (hunk 1 above)
Comment (uMartinXu) [reply docker/intel-dgpu-driver.Dockerfile:36]: Sorry, I didn't express my meaning clearly. 
The line is right.
But I'd like to know why the previous patch, it removes the line and then add the same line?
Maybe the two lines have some characters that can't be printed like space?

thanks!

Comment (hershpa) [reply docker/intel-dgpu-driver.Dockerfile:36]: image

Comment (hershpa) [reply docker/intel-dgpu-driver.Dockerfile:36]: There was no newline at the end of line 36 before.

Comment (uMartinXu) [inline docker/intel-dgpu-driver.Dockerfile:36]: we should keep this line. 
RUN depmod -b /opt
is useful actually. :-)

Comment (hershpa) [reply docker/intel-dgpu-driver.Dockerfile:36]: Line 36 which contains the depmod is not actually removed. It looks like it is removed but what actually happened is a newline was not present before on line 36. With the PR, a newline is being added to line 36. Take a look at the git diff again. :)

**Comment by mregmi** [comment]
LGTM

**Comment by chaitanya1731** [comment]
lgtm.

**Comment by hershpa** [comment]
@chaitanya1731, @uMartinXu, @mregmi, What if we set `LABEL release=${KERNEL_FULL_VERSION}`? No need to hardcode, we can leverage variable. :)

**Comment by chaitanya1731** [comment]
> @chaitanya1731, @uMartinXu, @mregmi, What if we set `LABEL release=${KERNEL_FULL_VERSION}`? No need to hardcode, we can leverage variable. :)

I like the idea of this. We can change it to `LABEL release=${KERNEL_FULL_VERSION}`

**Comment by hershpa** [comment]
> > @chaitanya1731, @uMartinXu, @mregmi, What if we set `LABEL release=${KERNEL_FULL_VERSION}`? No need to hardcode, we can leverage variable. :)
> 
> I like the idea of this. We can change it to `LABEL release=${KERNEL_FULL_VERSION}`

Updated PR, this sounds like the best approach.

**Comment by uMartinXu** [comment]
The patch looks good to me.

**Pull request merged**